ably/realtime_channel.go

```diff
@@ -770,10 +770,10 @@ func (c *RealtimeChannel) Modes() []ChannelMode {
 	return modes
 }
 
-func (c *RealtimeChannel) Params() ChannelParams {
+func (c *RealtimeChannel) Params() channelParams {
```

Comment:
This should return `map[string]string` to avoid returning an unexported type.

Reply:
It doesn't give any errors, we are already using it in the tests. We can use it as long as we don't need to declare the type. In go, ofc we don't need to declare type at lot of places. Also, I want to be explicit about `channelParam` type, rather than 
returning random map

Reply:
I know we _can_ do it, it's just pretty unfriendly to the user. The user doesn't even know what's a `channelParams` so they wouldn't know what to do with it. They also can't use it in declarations, so they can't pass it to a function of their own, or put it in a struct, or whatever else. In general it's a quirk and should be avoided.

> Also, I want to be explicit about `channelParam` type, rather than returning random map

What purpose does that serve to the consumers of the API?

Reply:
Makes sense, explicit DS is easy to identify and easy to use. Calling params means it's returning params anyways.